tests/EvmAcceptanceTests/contracts/ERC20isZRC2.sol

```diff
@@ -0,0 +1,157 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+pragma solidity >=0.7.0 <0.9.0;
+
+// Safe Math Interface.
+contract SafeMath {
+
+    function safeAdd(uint128 a, uint128 b) public pure returns (uint128 c) {
+        c = a + b;
+        require(c >= a);
+    }
+
+    function safeSub(uint128 a, uint128 b) public pure returns (uint128 c) {
+        require(b <= a);
+        c = a - b;
+    }
+
+    function safeMul(uint128 a, uint128 b) public pure returns (uint128 c) {
+        c = a * b;
+        require(a == 0 || c / a == b);
+    }
+
+    function safeDiv(uint128 a, uint128 b) public pure returns (uint128 c) {
+        require(b > 0);
+        c = a / b;
+    }
+}
+
+
+interface ERC20Interface {
+    function totalSupply() external view returns (uint128);
+    function balanceOf(address tokenOwner) external view returns (uint128);
+    function allowance(address tokenOwner, address spender) external view returns (uint128);
+    function approve(address tokenOwner, uint128 tokens) external returns (bool);
+    function transfer(address to, uint128 tokens) external returns (bool);
+    function transferFrom(address from, address to, uint128 tokens) external returns (bool);
+
+    function mint(address recipient, uint128 amount) external returns (bool);
+    function burn(address recipient, uint128 amount) external returns (bool);
+}
+
+contract ERC20isZRC2 is ERC20Interface, SafeMath {
+    address private _contract_owner;
+    address private _zrc2_address;
+
+    constructor(address zrc2_address) {
+        _contract_owner = msg.sender;
+        _zrc2_address = zrc2_address;
+    }
+
+    function totalSupply() external view returns (uint128) {
+        return _read_scilla_uint128("total_supply");
+    }
+
+    function balanceOf(address tokenOwner) external view returns (uint128) {
+        return _read_scilla_map_uint128("balances", tokenOwner);
+    }
+
+    function transfer(address to, uint128 tokens) external returns (bool) {
+        _call_scilla_two_args("Transfer", to, tokens);
+        return true;
+    }
+
+    function transferFrom(address from, address to, uint128 tokens) external returns (bool) {
+        _call_scilla_three_args("TransferFrom", from, to, tokens);
+        return true;
+    }
+
+    function allowance(address tokenOwner, address spender) external view returns (uint128) {
+        return _read_scilla_nested_map_uint128("allowances", tokenOwner, spender);
+    }
+
+    function approve(address spender, uint128 new_allowance) external returns (bool) {
+        uint128 current_allowance = _read_scilla_nested_map_uint128("allowances", msg.sender, spender);
+        if (current_allowance >= new_allowance) {
+            _call_scilla_two_args("DecreaseAllowance", spender, new_allowance);
```

Comment:
I think this wants to be `_call_scilla_two_args("DecreaseAllowance", spender, current_allowance-new_allowance` ?
(and likewise for `IncreaseAllowance`) ?

Reply:
`some_current_allowance <- allowances[_sender][spender];
  current_allowance = get_val some_current_allowance;
  new_allowance =
    let amount_le_allowance = uint128_le amount current_allowance in
      match amount_le_allowance with
      | True => builtin sub current_allowance amount
      | False => zero
      end;
  allowances[_sender][spender] := new_allowance;`
If I understand correctly, this happens within ZRC2 contract.

Reply:
You're right.